tests/components/cover/test_demo.py

```diff
@@ -0,0 +1,103 @@
+"""The tests for the Demo cover platform."""
+import unittest
+import homeassistant.util.dt as dt_util
+import logging
+
+from datetime import timedelta
+from homeassistant.components import cover
+from tests.common import get_test_home_assistant, fire_time_changed
+
+ENTITY_COVER = 'cover.living_room_window'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TestCoverDemo(unittest.TestCase):
+    """Test the Demo cover."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        self.assertTrue(cover.setup(self.hass, {'cover': {
+            'platform': 'demo',
+        }}))
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop down everything that was started."""
+        self.hass.stop()
+
+    def test_close_cover(self):
+        """Test closing the cover."""
+        state = self.hass.states.get(ENTITY_COVER)
+        self.assertEqual(70, state.attributes.get('current_position'))
+        cover.close_cover(self.hass, ENTITY_COVER)
+        _LOGGER.info("time=%s", dt_util.utcnow())
```

Comment:
I saw - I am going to pull down your code an try to figure this out later today, unless you get it before I do.